packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java

```diff
@@ -27,82 +100,132 @@
-    this.registrar = registrar;
-    this.delegate = delegate;
-    this.activityLifecycleCallbacks =
-        new Application.ActivityLifecycleCallbacks() {
-          @Override
-          public void onActivityCreated(Activity activity, Bundle savedInstanceState) {}
+  @Override
+  public void onDetachedFromEngine(FlutterPluginBinding binding) {
+    pluginBinding = null;
+  }
 
-          @Override
-          public void onActivityStarted(Activity activity) {}
+  @Override
+  public void onAttachedToActivity(ActivityPluginBinding binding) {
+    activityBinding = binding;
+    setup(
+        pluginBinding.getBinaryMessenger(),
+        (Application) pluginBinding.getApplicationContext(),
+        activityBinding.getActivity(),
+        null,
+        activityBinding);
+  }
 
-          @Override
-          public void onActivityResumed(Activity activity) {}
+  @Override
+  public void onDetachedFromActivity() {
+    tearDown();
+  }
 
-          @Override
-          public void onActivityPaused(Activity activity) {}
+  @Override
+  public void onDetachedFromActivityForConfigChanges() {
+    onDetachedFromActivity();
+  }
 
-          @Override
-          public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
-            if (activity == registrar.activity()) {
-              delegate.saveStateBeforeResult();
-            }
-          }
-
-          @Override
-          public void onActivityDestroyed(Activity activity) {
-            if (activity == registrar.activity()
-                && registrar.activity().getApplicationContext() != null) {
-              ((Application) registrar.activity().getApplicationContext())
-                  .unregisterActivityLifecycleCallbacks(
-                      this); // Use getApplicationContext() to avoid casting failures
-            }
-          }
-
-          @Override
-          public void onActivityStopped(Activity activity) {}
-        };
-
-    if (this.registrar != null
-        && this.registrar.context() != null
-        && this.registrar.context().getApplicationContext() != null) {
-      ((Application) this.registrar.context().getApplicationContext())
-          .registerActivityLifecycleCallbacks(
-              this
-                  .activityLifecycleCallbacks); // Use getApplicationContext() to avoid casting failures.
+  @Override
+  public void onReattachedToActivityForConfigChanges(ActivityPluginBinding binding) {
+    onAttachedToActivity(binding);
+  }
+
+  private void setup(
+      final BinaryMessenger messenger,
+      final Application applicationContext,
+      final Activity activity,
+      final PluginRegistry.Registrar registrar,
+      final ActivityPluginBinding activityBinding) {
+    this.activity = activity;
+    this.applicationContext = applicationContext;
+    this.delegate = constructDelegate(activity);
+    channel = new MethodChannel(messenger, CHANNEL);
+    channel.setMethodCallHandler(this);
+    if (registrar != null) {
```

Comment:
nit: some of the statements in the below if/then blocks are the same, like `observer =`. It would be a little clearer if they were pulled out and the only things in this block were the ones that needed to be different.